### PR TITLE
ref(sb): Updates to Storybook colors page

### PR DESCRIPTION
**Pull request by vuluongj20**
A few updates to the Colors page:
 - Use color values directly from theme.tsx
 - Add link to Typography page at the end of docs page
 - ColorSwatch labels now default to `black` instead of `gray500`

**On that last item:** `gray500` is now a dynamic color that changes values based on the current theme. It no longer means "dark text color". Instead, we should use `black`, which is static and independent of theme.
Before:
Screen Shot 2021-11-29 at 5 07 04 PM

After:
Screen Shot 2021-11-29 at 5 08 29 PM

### CODE REVIEW

File: static/app/utils/theme.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ const lightColors = {
   pink100: 'rgba(250, 51, 150, 0.1)',
 };
 
-const darkColors = {
+export const darkColors = {
```
Comment (vuluongj20) [inline static/app/utils/theme.tsx:53]: @billyvg Think you had some concerns about exporting raw, un-aliased colors from `theme.tsx`?

Comment (billyvg) [reply static/app/utils/theme.tsx:53]: @vuluongj20 Yeah I think we want to avoid the situation where people import and use them directly.

This might be ok because:
1) you're only using this for storybook
2) it's not exposed via our theme

Here are some additional things we could do - 

Good:
Add a comment for both of these exports to warn 

Better:
Add a lint rule to disable importing `lightColors` and `darkColors`  from `theme.tsx` (this should be relatively easy to do with https://eslint.org/docs/rules/no-restricted-imports)

Best:
Add a lint rule to disable importing from `theme.tsx` at all (this is difficult since we import from `theme.tsx` a lot, and out of scope for this PR)

Comment (billyvg) [reply static/app/utils/theme.tsx:53]: Let me know if you are interested in adding a lint rule, I can help you if needed.

Comment (vuluongj20) [reply static/app/utils/theme.tsx:61]: Added some comments, and created a PR with import lint rules: https://github.com/getsentry/eslint-config-sentry/pull/119

File: docs-ui/stories/core/colors/tables.tsx [hunk 2]
```diff
@@ -6,15 +6,9 @@ import space from 'sentry/styles/space';
...
-type ColorDefinition = {
-  name: string;
-  lightValue: string;
-  darkValue: string;
-};
-
 type ColorGroup = {
   id: string;
-  colors: ColorDefinition[];
+  colors: string[];
```
Comment (billyvg) [inline docs-ui/stories/core/colors/tables.tsx:11]: Should this be `Array<keyof typeof darkTheme | keyof typeof lightTheme>`?

File: package.json [hunk 3]
```diff
@@ -162,7 +162,7 @@
     "babel-plugin-dynamic-import-node": "^2.2.0",
     "csstype": "^3.0.8",
     "eslint": "5.11.1",
-    "eslint-config-sentry-app": "1.67.0",
+    "eslint-config-sentry-app": "1.69.0",
```
Comment (billyvg) [inline package.json:165]: @vuluongj20 let's do this in a separate PR - you'll want to do `yarn add eslint-config-sentry-app@1.69.0` because you'll need to update the yarn lockfile as well. We also need to do this on the getsentry repo.

File: docs-ui/stories/core/colors/tables.tsx (hunk 2 above)
Comment (billyvg) [reply docs-ui/stories/core/colors/tables.tsx:11]: @vuluongj20 ^

Comment (vuluongj20) [reply docs-ui/stories/core/colors/tables.tsx]: Good point! Just updated this

**Review by billyvg — APPROVED** [review]
